commit 08353f264f67cdeb6088bd4f87f4c83817b4247f
Author: Mark Thomas <markt@apache.org>
Date:   Wed Apr 27 08:23:58 2016 +0000

    Review of Map / ConcurrentMap / ConcurrentHashMap
    Simplify. These fields are only used internally so they can be declared as Map with Java 8.
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@1741157 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/javax/el/ExpressionFactory.java b/java/javax/el/ExpressionFactory.java
index 664f623..ac0ec5e 100644
--- a/java/javax/el/ExpressionFactory.java
+++ b/java/javax/el/ExpressionFactory.java
@@ -34,7 +34,6 @@ import java.security.PrivilegedAction;
 import java.util.Map;
 import java.util.Properties;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
@@ -56,8 +55,7 @@ public abstract class ExpressionFactory {
     private static final String PROPERTY_FILE;
 
     private static final CacheValue nullTcclFactory = new CacheValue();
-    private static final ConcurrentMap<CacheKey, CacheValue> factoryCache =
-            new ConcurrentHashMap<>();
+    private static final Map<CacheKey, CacheValue> factoryCache = new ConcurrentHashMap<>();
 
     static {
         if (IS_SECURITY_ENABLED) {
diff --git a/java/javax/el/Util.java b/java/javax/el/Util.java
index 9338835..5d6b4f5 100644
--- a/java/javax/el/Util.java
+++ b/java/javax/el/Util.java
@@ -31,7 +31,6 @@ import java.util.MissingResourceException;
 import java.util.ResourceBundle;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
@@ -80,8 +79,7 @@ class Util {
 
 
     private static final CacheValue nullTcclFactory = new CacheValue();
-    private static final ConcurrentMap<CacheKey, CacheValue> factoryCache =
-            new ConcurrentHashMap<>();
+    private static final Map<CacheKey, CacheValue> factoryCache = new ConcurrentHashMap<>();
 
     /**
      * Provides a per class loader cache of ExpressionFactory instances without
